### ISSUE TITLE
Request 'https://ipinfo.io/' timeout 

### ISSUE THREAD

**Issue by wy-z** [mobilecoinfoundation/mobilecoin#583]
Hi everyone, your work is amazing!
Uh, it seems that Mobilecoind will access "ipinfo.io", but "ipinfo.io" is blocked somewhere.

```
thread 'main' panicked at 'Could not validate host: Reqwest(reqwest::Error { kind: Request, url: "https://ipinfo.io/111.197.233.95/json/", source: TimedOut })', mobilecoind/src/bin/main.rs:22:32
```
![image](https://user-images.githubusercontent.com/4213483/101508903-281f4800-39b3-11eb-96f3-e9753c4ee9f6.png)

**Comment by posix4e** [comment]
Are you sure it's not just you worked fine for me
```
posix4e@penguin:~/src/mobilecoin$ curl https://ipinfo.io/111.197.233.95/json/
{
  "ip": "111.197.233.95",
  "city": "Beijing",
  "region": "Beijing",
  "country": "CN",
  "loc": "39.9075,116.3972",
  "org": "AS4808 China Unicom Beijing Province Network",
  "timezone": "Asia/Shanghai",
  "readme": "https://ipinfo.io/missingauth"
```
Suggest we close this ticket

**Comment by wy-z** [comment]
![image](https://user-images.githubusercontent.com/4213483/101670748-8836ec80-3a8e-11eb-88f4-ba1e23b507d6.png)

**Comment by wy-z** [comment]
![image](https://user-images.githubusercontent.com/4213483/101670907-bfa59900-3a8e-11eb-8a36-5be4f4882041.png)

**Comment by wy-z** [comment]
![image](https://user-images.githubusercontent.com/4213483/101671060-f7acdc00-3a8e-11eb-913f-284b7a773434.png)

**Comment by wy-z** [comment]
@posix4e
It works in some places and not in others, it seems unstable.

**Comment by posix4e** [comment]
Yep! Do you recommend and alternative? Also thanks for the ticket!--
Alex Newman
404.507.6749

**Comment by wy-z** [comment]
I'm sorry, I don't have the right advice.
I configured environment variables for `mobilecoind`, but this is not suitable for normal people.
Thanks for your work again. @posix4e

**Comment by posix4e** [comment]
@wy-z no worries, i'll keep looking. For now, I will not recommend that you use a VPN. I would never recommend a VPN or Tor.

**Comment by posix4e** [comment]
Perhaps we close this for now?

**Comment by sugargoat** [comment]
We are still working on a better solution for this - we know this is a problem in certain geographies that don't support ipinfo.

**Comment by wy-z** [comment]
> Perhaps we close this for now?

Ok, thanks for your work! @posix4e @sugargoat

**Comment by jcape** [comment]
@integral-llc I'd like to use this issue to kill two birds with one stone: We should make a new crate, `mc-compliance-check` (or something), that can live in it's own repo.

The crate should have a couple axis of features:
- [ ] Blocklists, starting with `us-ofac` that uses our current `mobilecoind` rules.
- [ ] Location providers, starting with `ipinfo`, but we should add more as fallback (and randomly determine the fallback order when called) 

Both `us-ofac` and `ipinfo` should be built as default-features for this crate, but the location-check should always succeed when build with `no-default-features` (e.g. when performing unit tests).

The repo itself should perform it's own unit tests using GHA, and the crate should be published to crates.io.

**Comment by jgreat** [comment]
We now have an ipinfo.io account so we can avoid rate limits in testing/prod. Would like to make sure that we can use the `MC_IP_INFO_TOKEN` environment variable (if provided) to set the authentication header. See current mobilecoind implementation for details.
